src/NUnitConsole/nunit3-console.tests/ResultReporterTests.cs

```diff
@@ -96,12 +91,26 @@ public void ReportSequenceTest()
         }
 
         [Test]
+        public void CheckReportSettingsTest()
+        {
+            var expected = new [] {
+                "Run Settings",
+                "    Setting1Name: Setting1Value",
+                "    Setting2Name: Setting2Value",
+                ""
+            };
+
+            var actualSettingsReport  = GetReportLines(_reporter.WriteRunSettingsReport);
+            Assert.That(expected, Is.EqualTo(actualSettingsReport));
+        }
+
+        [Test]
         public void SummaryReportTest()
         {
-            var expected = new string[] {
+            var expected = new [] {
```

Comment:
You have to ask yourself "What level of C# is required to use this construct?" and "Will it eliminate some people from building NUnit because they have an older version?"

In fact, we have a pending issue to document what level we support but we generally try to stick with C# 3.0 for most purposes. 

Not saying you are wrong here, just that I'm not sure what the impact will be.